azkaban-common/src/main/java/azkaban/trigger/Trigger.java

```diff
@@ -190,8 +191,17 @@ public static Trigger fromJson(final Object obj) throws Exception {
   }
 
   public void updateNextCheckTime() {
+    // expireCondition's nextCheckTime is the pre-set value, for most cases it's DEFAULT_SCHEDULE_END_EPOCH_TIME,
+    // It is not truly reflecting the real expiration time, so we need to update trigger manually.
     this.nextCheckTime = Math.min(this.triggerCondition.getNextCheckTime(),
         this.expireCondition.getNextCheckTime());
+    // manually update trigger's expiration status based on whether the nextCheckTime is before now.
+    // As nextCheckTime is supposed to get the next valid time, so if it's before now, it means the trigger is expired.
+    final DateTimeZone timezone = DateTimeZone.getDefault();
+    final DateTime date = new DateTime(this.nextCheckTime).withZone(timezone);
+    if (date.isBeforeNow()) {
```

Comment:
Do we need to worry about DST here? If next execution is at 01:30, and current time is 01:59 PDT, we will mark it as expired, and then one minute later we "fall back" to 01:00 PST (not PDT anymore) and so we should be executing it in 30 minutes.. I think?

Reply:
it's fine. The nextCheckTime we calculated, will only based on PST, it will not give us the 01:30PDT.